File: doc/src/scripts/tutorials/reference-frames/NavballSpeed.cs
using System;
using KRPC.Client;
using KRPC.Client.Services.SpaceCenter;

class NavballSpeed
{
    public static void Main ()
    {
        var connection = new Connection (name : "Navball speed");
        var vessel = connection.SpaceCenter ().ActiveVessel;
        var obtFrame = vessel.OrbitSpeedReferenceFrame;
        var srfFrame = vessel.SurfaceSpeedReferenceFrame;
        while (true) {
            var obtSpeed = vessel.Flight (obtFrame).Speed;
            var srfSpeed = vessel.Flight (srfFrame).Speed;
            Console.WriteLine (
                "Orbital speed = {0:F1} m/s, Surface speed = {1:F1} m/s",
                obtSpeed, srfSpeed);
            System.Threading.Thread.Sleep (1000);
        }
    }
}

File: doc/src/scripts/tutorials/reference-frames/NavballSpeed.py
import time
import krpc

conn = krpc.connect(name="Navball speed")
vessel = conn.space_center.active_vessel
obt_frame = vessel.orbit_speed_reference_frame
srf_frame = vessel.surface_speed_reference_frame

while True:
    obt_speed = vessel.flight(obt_frame).speed
    srf_speed = vessel.flight(srf_frame).speed
    print("Orbital speed = %.1f m/s, Surface speed = %.1f m/s" % (obt_speed, srf_speed))
    time.sleep(1)
